### DISCUSSION TITLE
SMTP error from two-step login page

### DISCUSSION THREAD

**Discussion by jgroll** [dani-garcia/vaultwarden#1685] [Q&A] [answered]
Hi 👋 
I have a problem with sending mail from the two-step login page for mailing.  When I want to activate the email option I get an error and the email with the activation code is not sent. 

Here the logs from docker : 
```
vaultwarden    | [2021-05-11 10:37:46.119][vaultwarden::mail][ERROR] SMTP 5xx error: lettre::transport::smtp::Error { kind: Permanent(Code { severity: PermanentNegativeCompletion, category: MailSystem, detail: Zero }) }
vaultwarden    | [2021-05-11 10:37:46.120][response][INFO] POST /api/two-factor/send-email (send_email) => 400 Bad Request
```
Mails work in all other cases:
- to send invitations
- login notification
- test mail from the admin page

Here some informations from diagnostics page 
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.21.0
* Web-vault version: v2.19.0d
* Running within Docker: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.33.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******.******.***",
  "domain_origin": "*****://******.******.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_folder": "data/sends",
  "show_password_hint": true,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "*****@******.**",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****.*****.***",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "*****@******.**",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```

Does anyone have an idea how to solve this problem?
Thank you 🙏

**Comment by BlackDex** [comment]
Well, the 5xx error is coming from the mail server, so something is going wrong there.
Try setting `SMTP_DEBUG=true` and use the `/admin` `smtp test` function interface under the `SMTP/Mail Settings`.
It should provide a bit more information, but maybe not enough. For more information you probably need to see the logs on the mail-server side. But it could be just enough to give  you a hint on where to look :).

**Reply by jgroll** [reply]
Thanks for your help. 
I am now thinking of a problem on the reverse proxy side (nginx). I have deployed vaultwarden locally and it works. Do you know anything about a reverse proxy that can cause this error ? Maybe a HTTP version ? On the reverse proxy I'm using HTTP 2.

**Comment by jgroll** [comment, marked as answer]
I finally found the error. The anti-spam filter of the mail server was blocking this exact request. 
`The email is flagged as a phishing email with brand impersonation`

Just allow these requests and it will fixed.
